exercises/e001/results/propiro/outputBase32faster.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+
+import string
+import random
+import argparse
+
+parser = argparse.ArgumentParser(description='base32 key generator script')
+parser.add_argument('size', const=32, default=32, type=int, help='size of key, default=32', nargs='?')
+caseGroup = parser.add_mutually_exclusive_group()
+caseGroup.add_argument('-u', action="store_true", help='use uppercase letters, default option', default=True)
+caseGroup.add_argument('-l', action="store_true", help='use lowercase letters', default=False)
+
+arg = parser.parse_args()
+
+arg.u = not arg.l
+
+def generateString(keySize =arg.size, charCase = arg.u, ):
```

Comment:
This is not typical pattern. I understand this works, but I would write it differently. Declare function just as:

```
def generateString(keySize, charCase):
```

and then later in the code you call this function as:

```
generateString(arg.size, arg.u)
```